Pet-Clinic-Frontend/src/components/AddNewPetForm.tsx

```diff
@@ -1,17 +1,53 @@
-import { useState } from "react"
+import { type Dispatch, type SetStateAction, useState } from "react"
+import { useNavigate } from "react-router-dom"
+import { useAuth } from "../pages/AuthContext"
+
+const backendUrl = import.meta.env.VITE_BACKEND_URL || "http://localhost:8080"
 
 type Props = {
-  setShowForm: (value: boolean) => void
+  setRenderForm: (value: boolean) => void
+  setParentError: Dispatch<SetStateAction<string | null>>
 }
 
-export const AddNewPetForm = ({ setShowForm }: Props) => {
+export const AddNewPetForm = ({ setRenderForm, setParentError }: Props) => {
   const [name, setName] = useState("")
   const [type, setType] = useState("")
+  const navigate = useNavigate()
+  const { token } = useAuth()
+
+  const validateForm = () => {
+    if (!name || !type) {
+      setParentError("Name and type are required")
+      return false
+    }
+    return true
+  }
+  const sendToBackend = async (name: string, type: string) => {
+    try {
+      const response = await fetch(`${backendUrl}/api/pets/add`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${token}`,
+        },
+        body: JSON.stringify({ name, type }),
+      })
```

Comment:
After the httpCLient utility is merged we should aggregate/use our http calls there.